index.js

```diff
@@ -23,11 +25,13 @@ function resetCaches() {
 }
 
 // Take a unstripped parsed document (query/mutation or even fragment), and
-// check all fragment definitions, checking for name->source uniqueness
+// check all fragment definitions, checking for name->source uniqueness.
+// We also want to make sure only unique fragments exist in the document.
 var printFragmentWarnings = true;
 function checkFragments(ast) {
```

Comment:
I think now that this function actually makes modifications to the AST, it should be called something else, like `processFragments`. It just struck me that `parsed = checkFragments(parsed)` looks weird since why would `checkFragments` have a return value?